### PR TITLE
implement AuthenticationController, OrderController, ShoppingCartCont…

### PR DESCRIPTION
**Pull request by IwalkInSilence**
…roller, mappers, dto for entity

### CODE REVIEW

**Review by YuraMitriakhin — APPROVED** [review]

File: src/main/java/project/cinema/controller/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import project.cinema.model.dto.UserRequestDto;
+import project.cinema.security.AuthenticationService;
+
+@RestController
+public class AuthenticationController {
+    private AuthenticationService authenticationService;
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/controller/AuthenticationController.java]:
```suggestion
    private final AuthenticationService authenticationService;
```
let's add `final` to all autowired fields

File: src/main/java/project/cinema/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    public void completeOrder(@RequestParam Long id) {
+        orderService.completeOrder(shoppingCartService.getByUser(userService.getById(id).get()));
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> getOrders(@RequestParam Long id) {
+        List<Order> ordersHistory = orderService.getOrdersHistory(userService.getById(id).get());
+        return ordersHistory
+                .stream()
+                .map(o -> orderMapper.parseToDto(o))
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/controller/OrderController.java]: use method reference

File: src/main/java/project/cinema/controller/ShoppingCartController.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.web.bind.annotation.RestController;
+import project.cinema.model.ShoppingCart;
+import project.cinema.model.dto.ShoppingCartResponseDto;
+import project.cinema.service.MovieSessionService;
+import project.cinema.service.ShoppingCartMapper;
+import project.cinema.service.ShoppingCartService;
+import project.cinema.service.UserService;
+
+@RestController
+@RequestMapping("/shopping-cart")
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/controller/ShoppingCartController.java]:
```suggestion
@RequestMapping("/shopping-carts")
```

File: src/main/java/project/cinema/dao/MovieSessionDao.java [hunk 4]
```diff
@@ -12,4 +12,6 @@
     void update(MovieSession movieSession);
 
     void delete(Long id);
+
+    MovieSession getById(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/dao/MovieSessionDao.java]: let's return Optional here as well

File: src/main/java/project/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 5]
```diff
@@ -102,4 +102,13 @@ public void delete(Long id) {
             }
         }
     }
+
+    @Override
+    public MovieSession getById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.get(MovieSession.class, id);
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/dao/impl/MovieSessionDaoImpl.java]:
```suggestion
            return Optional.ofNullable(session.get(MovieSession.class, id));
```

File: src/main/java/project/cinema/model/dto/OrderResponseDto.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
+package project.cinema.model.dto;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class OrderResponseDto {
+    private Long id;
+    private List<Long> ticketsIdList;
+    private LocalDateTime orderDate;
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/model/dto/OrderResponseDto.java:9]:
```suggestion
    private String orderDate;
```

File: src/main/java/project/cinema/service/UserService.java [hunk 7]
```diff
@@ -7,4 +7,6 @@
     User add(User user);
 
     Optional<User> findByEmail(String email);
+
+    Optional<User> getById(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/service/UserService.java]:
```suggestion
    User getById(Long id);
```
let's return the entity from this method

File: src/main/java/project/cinema/service/impl/mappers/OrderMapperImpl.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+import project.cinema.service.OrderMapper;
+
+@Component
+public class OrderMapperImpl implements OrderMapper {
+    @Override
+    public OrderResponseDto parseToDto(Order order) {
+        OrderResponseDto dto = new OrderResponseDto();
+        dto.setId(order.getId());
+        dto.setOrderDate(order.getOrderDate());
+        dto.setUserId(order.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/service/impl/mappers/OrderMapperImpl.java]: is this line ok?

File: src/main/java/project/cinema/service/impl/mappers/ShoppingCartMapperImpl.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Controller;
+import project.cinema.model.ShoppingCart;
+import project.cinema.model.dto.ShoppingCartResponseDto;
+import project.cinema.service.ShoppingCartMapper;
+
+@Controller
+public class ShoppingCartMapperImpl implements ShoppingCartMapper {
+
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/service/impl/mappers/ShoppingCartMapperImpl.java]: redundant empty line

File: src/main/java/project/cinema/service/impl/mappers/ShoppingCartMapperImpl.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    @Override
+    public ShoppingCartResponseDto parseToDto(ShoppingCart shoppingCart) {
+        ShoppingCartResponseDto dto = new ShoppingCartResponseDto();
+        dto.setId(shoppingCart.getId());
+        dto.setTicketsIdList(shoppingCart.getTickets()
+                 .stream()
+                 .map(t -> t.getId())
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/service/impl/mappers/ShoppingCartMapperImpl.java]: let's use method reference

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/project/cinema/model/dto/OrderResponseDto.java (hunk 6 above)
Comment (IwalkInSilence) [reply src/main/java/project/cinema/model/dto/OrderResponseDto.java:9]: Bohdan said that better to send LocalDateTime 
image

File: src/main/java/project/cinema/service/UserService.java (hunk 7 above)
Comment (IwalkInSilence) [reply src/main/java/project/cinema/service/UserService.java]: fix

File: src/main/java/project/cinema/dao/impl/MovieSessionDaoImpl.java (hunk 5 above)
Comment (IwalkInSilence) [reply src/main/java/project/cinema/dao/impl/MovieSessionDaoImpl.java]: fix

File: src/main/java/project/cinema/dao/MovieSessionDao.java (hunk 4 above)
Comment (IwalkInSilence) [reply src/main/java/project/cinema/dao/MovieSessionDao.java]: fix

File: src/main/java/project/cinema/controller/OrderController.java (hunk 2 above)
Comment (IwalkInSilence) [reply src/main/java/project/cinema/controller/OrderController.java]: fix

File: src/main/java/project/cinema/controller/AuthenticationController.java (hunk 1 above)
Comment (IwalkInSilence) [reply src/main/java/project/cinema/controller/AuthenticationController.java]: fix

File: src/main/java/project/cinema/dao/UserDao.java [hunk 11]
```diff
@@ -7,4 +7,6 @@
     User add(User user);
 
     Optional<User> findByEmail(String email);
+
+    User getById(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/dao/UserDao.java]:
```suggestion
    Optional<User> getById(Long id);
```
let's return Optional on dao layer - and User on service layer

File: src/main/java/project/cinema/service/MovieSessionService.java [hunk 12]
```diff
@@ -12,4 +13,6 @@
     void update(MovieSession movieSession);
 
     void delete(Long id);
+
+    Optional<MovieSession> getById(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/service/MovieSessionService.java]:
```suggestion
    MovieSession getById(Long id);
```
same - Optional from dao layer, entity from service layer

File: src/main/java/project/cinema/service/impl/MovieSessionServiceImpl.java [hunk 13]
```diff
@@ -34,4 +35,9 @@ public void update(MovieSession movieSession) {
     public void delete(Long id) {
         movieSessionDao.delete(id);
     }
+
+    @Override
+    public Optional<MovieSession> getById(Long id) {
+        return movieSessionDao.getById(id);
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/service/impl/MovieSessionServiceImpl.java]:
```suggestion
        return movieSessionDao.getById(id).get();
```

File: src/main/java/project/cinema/service/impl/UserServiceImpl.java [hunk 14]
```diff
@@ -28,4 +28,9 @@ public User add(User user) {
     public Optional<User> findByEmail(String email) {
         return userDao.findByEmail(email);
     }
+
+    @Override
+    public User getById(Long id) {
+        return userDao.getById(id);
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/service/impl/UserServiceImpl.java]:
```suggestion
        return userDao.getById(id).get();
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**